### PR TITLE
Internal: Support internal prop types [EDS-431]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: modules/atomic-widgets/prop-types/prop-type.php [hunk 1]
```diff
@@ -87,7 +87,7 @@ public function jsonSerialize(): array {
 		return [
 			'type' => [
 				'key' => static::get_key(),
-				'default' => $this->default,
+				'default' => $this->get_default(),
 				'settings' => (object) $this->settings,
```
Comment (Nevoss) [inline modules/atomic-widgets/prop-types/prop-type.php:91]:
```suggestion
				'settings' => (object) $this->get_settings(),
```

File: modules/atomic-widgets/props-resolver/settings-transformers/image-src-transformer.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		return Image_Src_Prop_Type::get_key();
+	}
+
+	// This transformer (or rather this prop type) exists only to support dynamic images.
+	// Currently, the dynamic tags that return images return it with id & url no matter
+	// what, so we need to keep the same structure in the props.
+	public function transform( $value ) {
+		return [
+			'id' => isset( $value['id'] ) ? (int) $value['id'] : null,
+			'url' => isset( $value['url'] ) ? esc_url( $value['url'] ) : null,
```
Comment (Nevoss) [inline modules/atomic-widgets/props-resolver/settings-transformers/image-src-transformer.php:23]: remove esc

File: modules/atomic-widgets/props-resolver/settings-transformers/image-transformer.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			);
+
+			if ( ! $image_src ) {
+				throw new \Exception( 'Cannot get image src.' );
+			}
+
+			[ $src, $width, $height ] = $image_src;
+
+			return [
+				'src' => esc_url( $src ),
```
Comment (Nevoss) [inline modules/atomic-widgets/props-resolver/settings-transformers/image-transformer.php:31]: remove all esc from tranformers
add `esc_url` in the `render` method of `atomic-image`

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: modules/atomic-widgets/dynamic-tags/dynamic-prop-types-mapping.php [hunk 4]
```diff
@@ -6,6 +6,7 @@
 use Elementor\Modules\AtomicWidgets\PropTypes\Number_Prop_Type;
 use Elementor\Modules\AtomicWidgets\PropTypes\Prop_Type;
 use Elementor\Modules\AtomicWidgets\PropTypes\String_Prop_Type;
+use Elementor\Modules\AtomicWidgets\PropTypes\Transformable_Prop_Type;
 use Elementor\Modules\DynamicTags\Module as V1_Dynamic_Tags_Module;
 
 class Dynamic_Prop_Types_Mapping {
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply modules/atomic-widgets/dynamic-tags/dynamic-prop-types-mapping.php:12]: Done

File: modules/atomic-widgets/prop-types/image-prop-type.php [hunk 5]
```diff
@@ -2,34 +2,70 @@
...
+				'src' => [
+					'$$type' => Image_Src_Prop_Type::get_key(),
+					'value' => [
+						'url' => $this->default_url,
+						'id' => $this->default_id,
+					],
+				],
+				'size' => $this->default_size,
+			],
+		];
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply modules/atomic-widgets/prop-types/image-prop-type.php:41]: Done

File: modules/atomic-widgets/prop-types/prop-type.php (hunk 1 above)
Comment (StyleShit) [reply modules/atomic-widgets/prop-types/prop-type.php:91]: Done

File: modules/atomic-widgets/props-resolver/settings-transformers/image-src-transformer.php (hunk 2 above)
Comment (StyleShit) [reply modules/atomic-widgets/props-resolver/settings-transformers/image-src-transformer.php:23]: Done

File: modules/atomic-widgets/props-resolver/settings-transformers/image-transformer.php (hunk 3 above)
Comment (StyleShit) [reply modules/atomic-widgets/props-resolver/settings-transformers/image-transformer.php:31]: Done

**Review by Nevoss — APPROVED** [review]

**Pull request merged**